### ISSUE TITLE
Toast not showing in some devices

### ISSUE THREAD

**Issue by ilyasdirin** [GrenderG/Toasty#110]
Hi, one of my clients with ASUS Z00AD, Android 6.0.1 told me that he is not able to see toast messages. I can not reproduce this situation in my test devices or emulators but he send me the video that does not show any toast message in where it has to. So is there any suggestion with this issue? (Used toasty version: 1.4.1)

**Comment by GrenderG** [comment]
That's weird, I have never encountered that issue.

**Comment by zwc456baby** [comment]
> That's weird, I have never encountered that issue.

If the user turns off the app's notification bar permission, toast will not be displayed on some devices, which is not a separate case. This problem may occur on Samsung or Huawei or other mobile phones, provided that the app's notification bar permissions are disabled.

**Comment by ilyasdirin** [comment]
> > That's weird, I have never encountered that issue.
> 
> If the user turns off the app's notification bar permission, toast will not be displayed on some devices, which is not a separate case. This problem may occur on Samsung or Huawei or other mobile phones, provided that the app's notification bar permissions are disabled.

Thats right, after user gave notification permission Toasty toasts started to show. Thanks for the reason. But the problem is there may be more users in production who disables notification permissions. So I guess there is nothing to do if the notification permissions are not allowed. Facebook, Instagram and a lot more apps also uses toast messages how do they handle this situation is there any way? Thanks

**Comment by zwc456baby** [comment]
> > > That's weird, I have never encountered that issue.
> > 
> > 
> > If the user turns off the app's notification bar permission, toast will not be displayed on some devices, which is not a separate case. This problem may occur on Samsung or Huawei or other mobile phones, provided that the app's notification bar permissions are disabled.
> 
> Thats right, after user gave notification permission Toasty toasts started to show. Thanks for the reason. But the problem is there may be more users in production who disables notification permissions. So I guess there is nothing to do if the notification permissions are not allowed. Facebook, Instagram and a lot more apps also uses toast messages how do they handle this situation is there any way? Thanks

You need to customize the toast tool without the permission of the notification bar. It is recommended to view the source code of this project.

[View project](https://github.com/getActivity/ToastUtils)

[View blog](https://www.jianshu.com/p/1d64a5ccbc7c)

**Comment by zwc456baby** [comment]
> > > > That's weird, I have never encountered that issue.
> > > 
> > > 
> > > If the user turns off the app's notification bar permission, toast will not be displayed on some devices, which is not a separate case. This problem may occur on Samsung or Huawei or other mobile phones, provided that the app's notification bar permissions are disabled.
> > 
> > 
> > Thats right, after user gave notification permission Toasty toasts started to show. Thanks for the reason. But the problem is there may be more users in production who disables notification permissions. So I guess there is nothing to do if the notification permissions are not allowed. Facebook, Instagram and a lot more apps also uses toast messages how do they handle this situation is there any way? Thanks
> 
> You need to customize the toast tool without the permission of the notification bar. It is recommended to view the source code of this project.
> 
> [View project](https://github.com/getActivity/ToastUtils)
> 
> [View blog](https://www.jianshu.com/p/1d64a5ccbc7c)

The Toast tool class for this project, which can be displayed without the permission of the notification bar.

**Comment by zwc456baby** [comment]
If there is no notification bar permission, it is recommended that this project be compatible.

**Comment by SatyamKharote** [comment]
i have given notification permission, but still Toast is not working....
